### ISSUE TITLE
Segfault in Database thread

### ISSUE THREAD

**Issue by kyx0r** [Eloston/ungoogled-chromium#1677]
**Describe the bug**
[54220.113568] Database thread[1511]: segfault at 0 ip 000055f1a3a5fa43 sp 00007fcab486ff80 error 6 in chromium[55f1952ea000+f0d8000]
[54220.113581] Code: d7 55 f6 48 8b 3b 48 83 ff 01 76 3b e8 56 e3 55 f6 49 89 c4 48 8b 43 08 48 89 45 a8 e8 86 d8 55 f6 49 8b 4c 24 18 48 8b 45 a8 <48> 89 08 48 c7 40 08 00 00 00 00 49 89 44 24 18 49 ff 4c 24 20 48

Segmentation Fault.

**To Reproduce**
Happens randomly. Sometimes chrome completely freezes the system
then it becomes necessary to reboot using the power button. Something
so bad that not even the kernel is not able to panic, and redraw past X11.
But I know for a fact that it does crash every other process on the system
to it's impossible to get any logs. This is the only clue I have.
If you want I can send you the elf executable for analysis. Sadly it is stripped and -march optimized
but you might be able to figure out what code crashed by reading the asm.

**Screenshots**
![2021-10-06-140917_1361x706_scrot](https://user-images.githubusercontent.com/33268383/136259438-71b9185f-2d1f-4074-9e3c-ef22c6f78b3e.png)

**Environment (please complete the following information):**
 - OS/Platform and version:
Linux 5.15.0-rc2+ x86_64 GNU/Linux
 - ungoogled-chromium version:
94.0.4606.71
exact build scripts used:
https://github.com/kyx0r/repo-community/tree/master/community/chromium

Most of the dependant libraries are kept up to date, any pointers to what this database thread is?

**Comment by kyx0r** [comment]
You are right, there are just simply way too many factors to consider here. I had that profile from version 87 I think, who knows what kind of corruption is going on there. I should get back into habbit of running chrome in some kind of container so that thing leaves no traces on my main system. I do run some exotic cpu and had some hardware faults related to cpu states where mobo supplies too low voltage to the cpu across the cores which causes it to randomly shut off, but I fixed it by disabling cpu states in bios, since then never seen those kinds of faults. Random system lockups (only when running chromium) is something else entirely.

**Comment by kyx0r** [comment]
Yes, currently using it, I deleted the .config/chromium folder and the .cache, so it should be clean. The issue still persists, in fact I notice that when it segfaults the page might change to oh snap something went wrong

**Comment by Ahrotahn** [comment]
You might also want to give the portable binaries a try for a while to see if the same thing happens with that build.  Or alternatively you could try a musl-based compiled build from another distro which might narrow down some build config differences.

**Comment by kyx0r** [comment]
I'll try building it with -g flag for debug info, but still the same optimization level, (because it is riskly to change opt level if we wanna get down to the root cause, compiler bugs can happen too) so I'll be able to see what function crashes exactly.

**Comment by wingedfox** [comment]
Well, this is the only recent bug report related to the segfault and the same happens to the other extensions but in a random manner. 
I'll report to Chromium as well, of course.
